src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Deployments.Core.Entities;
+using Azure.Deployments.Core.Helpers;
+using Azure.Deployments.Core.Json;
+using Bicep.Core;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Semantics;
+using Bicep.Core.Text;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // This handler is used to generate compiled .json file for given a bicep file path.
+    // It returns build succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepBuildCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager CompilationManager;
+
+        public BicepBuildCommandHandler(ICompilationManager compilationManager, ISerializer serializer)
+            : base(LanguageConstants.Build, serializer)
+        {
+            CompilationManager = compilationManager;
+        }
+
+        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string buildOutput = GenerateCompiledFileAndReturnBuildOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(buildOutput);
+        }
+
+        private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            string compiledFilePath = PathHelper.GetDefaultBuildOutputPath(bicepFilePath);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            try
+            {
+                // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+                // If not, we'll fail the build.
+                if (File.Exists(compiledFilePath) && !TemplateContainsBicepGeneratorMetadata(File.ReadAllText(compiledFilePath)))
+                {
+                    return "Build failed. The file \"" + compiledFile + "\" already exists and was not generated by Bicep. If overwriting the file is intended, delete it manually and retry the Build command.";
+                }
+            }
+            catch (JsonReaderException)
```

Comment:
>JsonReaderException [](http://example.com/codeflow?start=19&length=19)

Are we sure that the template library can't throw any other exceptions when the template is malformed? It seems like it'll be safer to just catch `Exception` and return false in `TemplateContainsBicepGeneratorMetadata()`. 

(We don't really need to give the user the reason why we think Bicep didn't generate the file.)

Reply:
As discussed offline, updated to catch exception and return false:
https://github.com/Azure/bicep/commit/79e7a2b9d06e87589572453b62ebb06cbaaab30f